File: app/src/main/java/com/spacex/launches/spacexrocketlaunches/EntryActivity.java
package com.spacex.launches.spacexrocketlaunches;

import android.app.Activity;
import com.spacex.launches.spacexrocketlaunches.rocketList.RocketListActivity;
import com.spacex.launches.spacexrocketlaunches.welcome.WelcomeActivity;

public class EntryActivity extends Activity {

  @Override
  protected void onStart() {
    super.onStart();
    showFirstVisibleActivity();
  }

  private void showFirstVisibleActivity() {
    if (InteractionsHistory.wasWelcomeShown()) {
      RocketListActivity.startActivity(this);
    } else {
      WelcomeActivity.startActivity(this);
    }
  }
}

File: app/src/main/kotlin/com/spacex/ui/base/EntryActivity.kt
package com.spacex.ui.base

import android.app.Activity
import com.spacex.ui.common.InteractionsHistory
import com.spacex.ui.rocketList.RocketListActivity
import com.spacex.ui.welcome.WelcomeActivity

internal class EntryActivity : Activity() {

    override fun onStart() {
        super.onStart()
        showFirstVisibleActivity()
    }

    private fun showFirstVisibleActivity() {
        if (InteractionsHistory.wasWelcomeShown()) {
            RocketListActivity.startActivity(this)
        } else {
            WelcomeActivity.startActivity(this)
        }
    }
}
